protocol-units/bridge/contracts/src/NativeBridge.sol

```diff
@@ -100,25 +134,44 @@ contract NativeBridge is AccessControlUpgradeable, PausableUpgradeable, INativeB
         uint256 amount,
         uint256 nonce
     ) internal {
-        // _l2l1RateLimit(amount);
+        _rateLimitInbound(amount);
+        // Ensure the bridge transfer has not already been completed
+        require(nonce > 0, InvalidNonce());
+        require(idsToInboundNonces[bridgeTransferId] == 0, CompletedBridgeTransferId());
         // Ensure the bridge transfer ID is valid against the initiator, recipient, amount, and nonce
         require(
             bridgeTransferId == keccak256(abi.encodePacked(initiator, recipient, amount, nonce)),
             InvalidBridgeTransferId()
         );
-        // Ensure the bridge transfer has not already been completed
-        require(noncesToIncomingBridgeTransferIds[nonce] == bytes32(0x0), CompletedBridgeTransferId());
 
         // Store the nonce to bridge transfer ID
-        noncesToIncomingBridgeTransferIds[nonce] = bridgeTransferId;
+        idsToInboundNonces[bridgeTransferId] = nonce;
 
         // Transfer the MOVE tokens to the recipient
         if (!moveToken.transfer(recipient, amount)) revert MOVETransferFailed();
 
         emit BridgeTransferCompleted(bridgeTransferId, initiator, recipient, amount, nonce);
     }
 
+    function setInsuranceFund(address _insuranceFund) external onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
I think this is better as a non-default  admin role. I would suggest `INSURANCE_ADMIN`.